examples/nodejs/lambda-examples/topics-microservice/lib/topics-microservice-stack.ts

```diff
@@ -0,0 +1,96 @@
+import {App, aws_secretsmanager, Duration, Stack} from 'aws-cdk-lib';
+import {RestApi, LambdaIntegration, Model, JsonSchemaType, RequestValidator} from 'aws-cdk-lib/aws-apigateway';
+import { NodejsFunction, NodejsFunctionProps } from 'aws-cdk-lib/aws-lambda-nodejs';
+import { Runtime } from 'aws-cdk-lib/aws-lambda';
+import { join } from 'path';
+
+export class TopicsMicroserviceStack extends Stack {
+  constructor(app: App, id: string) {
+    super(app, id);
+
+    const nodeJsFunctionProps: NodejsFunctionProps = {
+      environment: {
+        "RUNTIME": "AWS",
+        "REGION": "us-west-2",
+        "SECRETNAME": "Momento_Auth_Token",
+        "CACHENAME": "default-cache"
+      },
+      bundling: {
+        externalModules: [
+          'aws-sdk', // Use the 'aws-sdk' available in the Lambda runtime
+        ],
+      },
+      depsLockFilePath: join(__dirname, '..', 'package-lock.json'),
+      runtime: Runtime.NODEJS_18_X,
+      memorySize: 1024, // Increase memory to help with response times
+      timeout: Duration.seconds(10) // Make timeout longer for bootstrap api
+    }
+
+    // Create a Lambda function for handling service requests
+    const serviceLambda = new NodejsFunction(this, 'ServiceLambda', {
+      entry: join(__dirname, '../resources', 'topics.ts'),
+      ...nodeJsFunctionProps,
+    });
+
+    // Get the ARN for the existing secret the Lambda function will be using.
+    let secret = aws_secretsmanager.Secret.fromSecretNameV2(this, "Momento_Auth_Token", "Momento_Auth_Token");
```

Comment:
you have a very specific secret name hard-coded here, do you document that that is the required secret name somewhere else?

Reply:
Yeah, this is me not cleaning up when I was learning how to do this. so it is not taking in the passed in values. Thank you. I will fix this!

Reply:
Fixed.